src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -247,6 +247,14 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 				$theme_json_data = array( 'version' => WP_Theme_JSON::LATEST_SCHEMA );
 			}
 
+			// Register variations defined by the theme.
+			$variations = $theme_json_data['styles']['blocks']['variations'] ?? array();
+			wp_register_block_style_variations_from_theme_json_data( $variations );
+
+			// Register variations defined by theme partials (theme.json files in the styles directory).
+			$variations = self::get_style_variations( 'block' );
```

Comment:
```suggestion
			$variations = static::get_style_variations( 'block' );
```
Nit: Should we make this `static` rather than `self`?